react/ActionMenu/index.jsx

```diff
@@ -174,11 +176,12 @@ const ActionMenuItem = ({
   className,
   contentSpacing
 }) => {
+  const hasCaption = Array.isArray(children) && children.some(child => get(child, 'type.displayName', '').includes('Caption'))
```

Comment:
This seems too magical for me, and it links the two components (ActionMenuItem and Caption) too much I think. I ask myself what is the general rule here, and also what happens when we migrate to new text components. I think it'd be preferrable to let this responsibility to the app, by providing an align prop on the ActionMenuItem.

Reply:
The idea is to have a default behavior that vertically aligns the icon, except for cases with a caption, we then position the icon at the top.

Reply:
I understand the intention but I think that it may not be desirable to add this kind of coupling between Caption and ActionMenuItem.


* What happens when a new component other than Caption needs to have capability ? Should we keep adding new components in the list ?
* What happens if the Caption is wrapped into `<span></span>` to add for example another icon ? The check won't work 
* Is it really necessary to introduce a hard coded rule like that when it can be done by the app on a case by case basis ?

Reply:
Hmmm... you're right it's getting complicated, maybe we need to take a little more time to list all the cases and make a more complex rule? Or we can simply say that for the moment it's not relevant because there are no concrete cases, we're here in anticipation.

There are also questions with texts that are too long, and texts of different lengths in different languages... in this case, do you think we can manage on the application side?

Reply:
No more magic, this is not needed anymore